File: src/main/java/com/cfgglobal/test/cache/CacheClient.java
package com.cfgglobal.test.cache;


import java.util.Set;
import java.util.function.Supplier;

public interface CacheClient {

    <T> T get(String key);

    <T> T get(String key, Supplier<T> supplier);

    <T> T get(String key, Supplier<T> supplier, long expire);

    void set(String key, Object value);


    void set(String key, long expire, Object value);

    void deleteByKey(String key);

    void deleteByPattern(String key);

    Set<String> keys(String key);


}

File: src/main/kotlin/com/github/leon/cache/CacheClient.kt
package com.github.leon.cache


interface CacheClient {

    operator fun <T> get(key: String): T?

    operator fun <T> get(key: String, supplier: () -> T): T?

    operator fun <T> get(key: String, supplier: () -> T, expire: Long): T?

    operator fun set(key: String, value: Any)

    operator fun set(key: String, expire: Long, value: Any)

    fun deleteByKey(key: String)

    fun deleteByPattern(pattern: String)

    fun keys(key: String): Set<String>


}
